l1-contracts/src/core/messagebridge/NewOutbox.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Aztec Labs.
+pragma solidity >=0.8.18;
+
+// Libraries
+import {DataStructures} from "../libraries/DataStructures.sol";
+import {Errors} from "../libraries/Errors.sol";
+import {Merkle} from "../libraries/Merkle.sol";
+import {Hash} from "../libraries/Hash.sol";
+import {INewOutbox} from "../interfaces/messagebridge/INewOutbox.sol";
+
+/**
+ * @title NewOutbox
+ * @author Aztec Labs
+ * @notice Lives on L1 and is used to consume L2 -> L1 messages. Messages are inserted by the state transitioner
```

Comment:
Everywhere else in solidity we call state transitioner a Rollup so would prefer to stick to that naming. I see Rollup as an implementation of the abstract "state transitioner" from yellow paper.